### PR TITLE
Validate levels in controls

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- add a function which will check if a control does not include invalid level
- valid levels are defined in the policy, any other levels are invalid

#### Rationale:

Inspired while verifying #11417 

#### Review Hints:

- choose a policy which has defined levels, change a control to contain undefined level and build any product
- choose a policy which does not use level and define a level for a particular control, build any product

### CODE REVIEW

File: ssg/controls.py [hunk 1]
```diff
@@ -233,6 +233,28 @@ def check_all_rules_exist(self, existing_rules):
...
+        """
+        This function goes through all controls in the policy and checks if all
+        levels defined for individual controls are valid for the policy.
+        If the policy has no levels defined, then all controls should have the
+        "default" level defined (this is defined implicitly).
+        """
+        for c in self.controls:
+            levels = [lvl for lvl in c.levels if c.levels is not None]
+            if not self.levels_by_id:
+                if levels != ["default"]:
```
Comment (Mab879) [inline ssg/controls.py:246]: We cannot use f-strings since this code this needs to run on Python 2.7.

File: ssg/controls.py [hunk 2]
```diff
@@ -233,6 +233,28 @@ def check_all_rules_exist(self, existing_rules):
...
+                    msg = (f"Invalid levels {levels} used in control {c.id} "
```
Comment (Mab879) [inline ssg/controls.py:247]: We cannot use f-strings since this code this needs to run on Python 2.7.

File: ssg/controls.py [hunk 3]
```diff
@@ -233,6 +233,28 @@ def check_all_rules_exist(self, existing_rules):
...
+                            f"defined at {self.filepath}. Level must be set to \"default\".")
+                    raise ValueError(msg)
+                return
+            for lvl in levels:
+                if lvl not in self.levels_by_id:
+                    msg = (f"Invalid level {lvl} used in control {c.id} "
```
Comment (Mab879) [inline ssg/controls.py:253]: We cannot use f-strings since this code this needs to run on Python 2.7.

**Review by Mab879 — CHANGES_REQUESTED** [review]
LGTM, but please rewrite to that it works on Python 2.7 on CentOS 7.

**Comment by vojtapolasek** [comment]
@Mab879  I think I fixed everything I could. I consider extracting code from controls.py to satisfy the requirement on number of lines as out of scope.

**Comment by Mab879** [comment]
> @Mab879 I think I fixed everything I could. I consider extracting code from controls.py to satisfy the requirement on number of lines as out of scope.

Agreed. Thanks for the work. Reviewing.

**Review by Mab879 — APPROVED** [review]
LGTM.

Thanks for your work.

**Pull request merged**